frontend/src/LineGraph.res

```diff
@@ -99,8 +99,9 @@ module DataRow = {
   }
 
   let add_value = (row: t, value: value): t => {
-    Js.Array.push(value, row)->ignore
-    row
+    let newRow = row->Belt.Array.sliceToEnd(0)
+    Js.Array.push(value, newRow)->ignore
+    newRow
```

Comment:
Mutating the array was causing some errors due to data being inconsistent. My guess is that now that we use the `useIntersection` hook, the component is being rendered more often (at least every time it enters/leaves the viewport). And since we were mutating the array, old data were somehow merged with the new one, ending up in a weird state. Anyway, we should be careful with mutations and that's probably better to avoid them as much as possible.